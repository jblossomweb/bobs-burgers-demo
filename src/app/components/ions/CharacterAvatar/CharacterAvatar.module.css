.avatar {
  @apply border-2 border-palette-dark-blue my-4;
}

.avatar:hover {
  @apply border-palette-light-blue;
}

.avatar:active {
  @apply translate-y-0.5;
}

.wrapper {
  overflow: hidden;
  @apply w-28 h-28 bg-white
    flex flex-wrap flex-row
    items-center justify-center;
}

.image {
  @apply w-full;
}
