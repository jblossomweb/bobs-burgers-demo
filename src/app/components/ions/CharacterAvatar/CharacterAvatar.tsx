import React from 'react'
import { Link } from 'react-router-dom'
import type Character from 'app/models/Character'
import Avatar from 'app/components/atoms/Avatar'
import styles from './CharacterAvatar.module.css'

export interface Props {
  character: Character
}

const CharacterAvatar: React.FC<Props> = ({ character }) => (
  <Avatar className={styles.avatar}>
    <div className={styles.wrapper}>
      <Link to={`/characters/${character.id}`}>
        <img
          className={styles.image}
          src={character.image}
          alt={character.name}
        />
      </Link>
    </div>
  </Avatar>
)

export default CharacterAvatar
