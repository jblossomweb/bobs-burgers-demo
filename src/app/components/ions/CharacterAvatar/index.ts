export { default, type Props } from './CharacterAvatar'
