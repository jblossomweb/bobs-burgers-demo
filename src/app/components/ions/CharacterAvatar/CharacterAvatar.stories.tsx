import React from 'react'
import type { ComponentStory, ComponentMeta } from '@storybook/react'
import mockCharacter from 'app/models/__mocks__/character.json'

import CharacterAvatar, { type Props } from './CharacterAvatar'

export default {
  title: 'components/ions/CharacterAvatar',
  component: CharacterAvatar,
  argTypes: {
    character: { control: 'object' },
  },
} as ComponentMeta<typeof CharacterAvatar>

const Template: ComponentStory<typeof CharacterAvatar> = ({
  character,
}: Props) => <CharacterAvatar character={character} />

export const Sample = Template.bind({})
Sample.args = {
  character: mockCharacter,
}
