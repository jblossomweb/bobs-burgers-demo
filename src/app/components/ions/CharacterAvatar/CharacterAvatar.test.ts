import storySnaps from 'test/utils/storySnaps'
import * as stories from './CharacterAvatar.stories'

describe('app/components/ions/CharacterAvatar', () => {
  describe('stories', () => {
    storySnaps(stories)
  })
})
