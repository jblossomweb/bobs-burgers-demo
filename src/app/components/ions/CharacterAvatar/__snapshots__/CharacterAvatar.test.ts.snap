// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`app/components/ions/CharacterAvatar stories Sample matches markup snapshot 1`] = `
<DocumentFragment>
  <div
    class="avatar avatar"
    style="background-color: rgb(97, 218, 251);"
  >
    <div
      class="wrapper"
    >
      <a
        href="/characters/47"
      >
        <img
          alt="Robert \\"Bob\\" Belcher, Jr."
          class="image"
          src="https://bobsburgers-api.herokuapp.com/images/characters/47.jpg"
        />
      </a>
    </div>
  </div>
</DocumentFragment>
`;
